src/provider.ts

```diff
@@ -0,0 +1,93 @@
+import Provider, { Configuration } from "oidc-provider";
+import { Config } from "./config.js";
+import { createOidcClientFromConfig } from "./helper/createOidcClientFromConfig.js";
+import { claimsValidator } from "./validators/claimsValidator.js";
+import { vtrValidator } from "./validators/vtrValidator.js";
+import { ISSUER_VALUE } from "./constants/providerConfig.js";
+
+const createOidcProvider = async (): Promise<Provider> => {
+  const config: Configuration = {
+    features: {
+      claimsParameter: {
+        enabled: true,
+        assertClaimsParameter: claimsValidator,
+      },
+      backchannelLogout: { enabled: true },
+      userinfo: { enabled: true },
+      pushedAuthorizationRequests: { enabled: false },
+      requestObjects: {
+        request: true,
+        requestUri: false,
+        requireSignedRequestObject: false,
+        requireUriRegistration: false,
+      },
+      registration: {
+        enabled: false,
+      },
+    },
+    extraParams: {
+      vtr: vtrValidator,
+    },
+    routes: {
+      authorization: "/authorize",
+      token: "/token",
+      userinfo: "/userinfo",
+      end_session: "/logout",
+      registration: "/connect/register",
+      jwks: "/.well-known/jwks.json",
+    },
+    responseTypes: ["code"],
+    claims: {
+      email: ["email", "email_verified"],
+      phone: ["phone_number", "phone_number_verified"],
+      openid: ["sub"],
+      "https://vocab.account.gov.uk/v1/passport": null,
+      "https://vocab.account.gov.uk/v1/socialSecurityRecord": null,
+      "https://vocab.account.gov.uk/v1/coreIdentityJWT": null,
+      "https://vocab.account.gov.uk/v1/address": null,
+      "https://vocab.account.gov.uk/v1/inheritedIdentityJWT": null,
+      "https://vocab.account.gov.uk/v1/drivingPermit": null,
+      wallet_subject_id: null,
+    },
+    scopes: ["openid", "email", "phone"],
+    subjectTypes: ["public", "pairwise"],
+    clientAuthMethods: ["private_key_jwt", "client_secret_post"],
```

Comment:
should we leave client_secret_post out here if we're not enabling it elsewhere?

Reply:
Yeah that makes more sense, will remove!